lib/bundler/fetcher/downloader.rb

```diff
@@ -11,22 +11,22 @@ def initialize(connection, redirect_limit)
         @redirect_limit = redirect_limit
       end
 
-      def fetch(uri, options = {}, counter = 0)
+      def fetch(uri, headers = {}, counter = 0)
         raise HTTPError, "Too many redirects" if counter >= redirect_limit
 
-        response = request(uri, options)
+        response = request(uri, headers)
         Bundler.ui.debug("HTTP #{response.code} #{response.message} #{uri}")
 
         case response
-        when Net::HTTPSuccess, Net::HTTPNotModified
+        when Net::HTTPSuccess, Net::HTTPNotModified, Net::HTTPRequestedRangeNotSatisfiable
```

Comment:
this looks a little suspect to me -- why are we treating 416 responses as successful?

Reply:
Agreed.

@MaxLap - I can see what you're doing here, but I think it would be clearer to add retry logic to this method (i.e., set `headers["Accept-Encoding"] = "gzip"` and then call `fetch(..)` again in a new `when Net::HTTPRequestedRangeNotSatisfiable` case.

With that done, `lib/bundler/compact_index_client/updater.rb` should remain unchanged.

Reply:
Well, if you specify a range and that range is wrong, being told by the server that it was wrong is the normal expected answer in a way. 

But I do also agree with what you are saying, and that was my initial fix idea. I just found it clunky to have to alter the headers in a method & class that just receives them and pass them through. 

I'll do the change tonight.